src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpGeneralHeaders.cs

```diff
@@ -81,14 +81,25 @@ internal sealed class HttpGeneralHeaders
 
         internal static bool? GetTransferEncodingChunked(HttpHeaders parent, HttpGeneralHeaders? headers)
         {
-            if (parent.ContainsParsedValue(KnownHeaders.TransferEncoding.Descriptor, HeaderUtilities.TransferEncodingChunked))
+            if (parent.TryGetHeaderValue(KnownHeaders.TransferEncoding.Descriptor, out object? value))
             {
-                return true;
+                // Fast-path for the very common case where "chunked" is the only value.
+                if (value is string stringValue && stringValue.Equals("chunked", StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+
+                if (parent.ContainsParsedValue(KnownHeaders.TransferEncoding.Descriptor, HeaderUtilities.TransferEncodingChunked))
```

Comment:
Does ContainsParsedValue perform any other side-effects that this now skips?

Reply:
Not as it's currently written - if the value is just `"chunked"`, then that's exactly what you would have seen from non-validated or validated accesses later on anyway.

If we tried being smarter with the above check, e.g. "parse all the values without allocating and look for chunked", then you could introduce subtle differences where a non-validating access would still see the original, non-parsed value. But that is also the whole point of NonValidated so it'd be even more correct IMO.